app/models/unit_type.rb

```diff
@@ -1,3 +1,11 @@
 class UnitType < ApplicationRecord
-  belongs_to :tower
+  validates :description, :metreage, presence: true
+
+  validate :metreage_cannot_be_less_than_one
+
+  private
+
+  def metreage_cannot_be_less_than_one
+    errors.add(:metreage, 'Metragem não pode ser igual ou menor que zero') if metreage.present? && metreage <= 0
+  end
```

Comment:
Uma forma de fazer esse método ligeiramente mais legível é usando "guard clause":

```suggestion
  def metreage_cannot_be_less_than_one
    return unless metreage.present?
    
    errors.add(:metreage, 'não pode ser igual ou menor que zero') if metreage <= 0
  end
```
Observação: pelo que me lembro, não é necessário colocar "Metragem" na mensagem, porque o Rails vai usar o i18n para criar a mensagem completa. Seria bom testar isso na aplicação rodando e colocar print da tela no PR também. O teste do Rspec não vai detectar se a mensagem aparecer na tela assim "Metragem Metragem não pode ser igual ou menor que zero".

https://campuscode.com.br/conteudos/guard-clause-em-ruby

Reply:
Lembrei que também validações padrão de números do rails que poderiam ser utilizadas.